src/index.d.ts

```diff
@@ -388,6 +388,28 @@ export interface BadgeProps {
  */
 export class Badge extends React.Component<BadgeProps> {}
 
+/**
+ * withBadge Higher-Order Component
+ * @param value 
+ * @param options 
+ */
+export function withBadge(
+  /**
+   * Text value to be displayed by badge
+   */
+  value: React.ReactNode | (() => React.ReactNode),
+  /**
+   * Options to configure the badge
+   */
+  options: {,
+    bottom?: number,
+    left?: number,
+    right?: number,
+    top?: number,
+    hidden?: boolean,
+  } & BadgeProps,
+): <P extends {}>(WrappedComponent: React.ComponentType<P>) => React.ComponentType<P>;
```

Comment:
Remove the `extends` just use `<P = {}>`

See https://github.com/react-native-training/react-native-elements/blob/next/src/index.d.ts#L2083 for example